pkg/domain/domain.go

```diff
@@ -1182,7 +1182,7 @@ func (do *Domain) loadSchemaInLoop(ctx context.Context) {
 		case <-do.exit:
 			return
 		}
-		do.refreshMDLCheckTableInfo()
+		do.refreshMDLCheckTableInfo(ctx)
```

Comment:
```
goroutine 507 [select]:
github.com/tikv/client-go/v2/config/retry.(*Config).createBackoffFn.newBackoffFn.func2({0x70d3428, 0xc047076990}, 0xffffffffffffffff)
	/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20241223070848-fd950fcf9fcc/config/retry/config.go:204 +0x4fd
github.com/tikv/client-go/v2/config/retry.(*Backoffer).BackoffWithCfgAndMaxSleep(0xc070e36cf0, 0xc00169e990, 0xffffffffffffffff, {0x7099b00, 0xc0d8e1e198})
	/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20241223070848-fd950fcf9fcc/config/retry/backoff.go:195 +0x66d
github.com/tikv/client-go/v2/config/retry.(*Backoffer).Backoff(0xc070e36cf0, 0xc00169e990, {0x7099b00, 0xc0d8e1e198})
	/home/genius/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20241223070848-fd950fcf9fcc/config/retry/backoff.go:122 +0x23e
github.com/pingcap/tidb/pkg/store/driver/backoff.(*Backoffer).Backoff(0x4?, 0x42?, {0x7099b00?, 0xc0d8e1e198?})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/store/driver/backoff/backoff.go:55 +0x25
github.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleCopResponse(0xc0459213b0, 0xc0705d8438, 0x0, 0xc0f767c870, {0xc03ea91080, 0x170, 0x170}, 0xc0232d4880, 0xc0e7c80a80, 0x12d2c)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1541 +0x445
github.com/pingcap/tidb/pkg/store/copr.(*copIteratorWorker).handleTaskOnce(0xc0459213b0, 0xc0705d8438, 0xc0e7c80a80)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1425 +0x102b
github.com/pingcap/tidb/pkg/store/copr.(*liteCopIteratorWorker).liteSendReq(0xc0470769c0, {0x70d3428?, 0xc0470357d0?}, 0xc0abcdeb40)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1186 +0x1c5
github.com/pingcap/tidb/pkg/store/copr.(*copIterator).Next(0xc0abcdeb40, {0x70d3428?, 0xc0470357d0?})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/store/copr/coprocessor.go:1109 +0x1d7
github.com/pingcap/tidb/pkg/distsql.(*selectResult).fetchResp(0xc04abe4700, {0x70d3428, 0xc0470357d0})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/distsql/select_result.go:318 +0x82
github.com/pingcap/tidb/pkg/distsql.(*selectResult).Next(0xc04abe4700, {0x70d3428, 0xc0470357d0}, 0xc0edd1c7d0)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/distsql/select_result.go:384 +0x9d
github.com/pingcap/tidb/pkg/executor.(*tableResultHandler).nextChunk(0x70d3428?, {0x70d3428?, 0xc0470357d0?}, 0x1f?)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/executor/table_reader.go:612 +0xa2
github.com/pingcap/tidb/pkg/executor.(*TableReaderExecutor).Next(0xc259214288, {0x70d3428, 0xc0470357d0}, 0xc0edd1c7d0)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/executor/table_reader.go:331 +0x213
github.com/pingcap/tidb/pkg/executor/internal/exec.Next({0x70d3428, 0xc0470357d0}, {0x710b080, 0xc259214288}, 0xc0edd1c7d0)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:460 +0x29f
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).next(0xc04abe4540, {0x70d3428, 0xc0470357d0}, {0x710b080, 0xc259214288}, 0xc0edd1c7d0)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/executor/adapter.go:1269 +0x6e
github.com/pingcap/tidb/pkg/executor.(*recordSet).Next(0xc045921420, {0x70d3428?, 0xc0470357d0?}, 0xc0edd1c7d0)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/executor/adapter.go:172 +0x108
github.com/pingcap/tidb/pkg/session.drainRecordSet({0x70d3428, 0xc0470357d0}, 0xc07ff386c8, {0x70d3ed8, 0xc02aa83040}, {0x0?, 0x0?})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/session/session.go:1238 +0xe4
github.com/pingcap/tidb/pkg/session.(*session).ExecRestrictedSQL.func1({0x70d3428, 0xc0470354d0}, 0xc07ff386c8)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/session/session.go:1981 +0x2d1
github.com/pingcap/tidb/pkg/session.(*session).withRestrictedSQLExecutor(0xc06e9a8248, {0x70d3428, 0xc0470354d0}, {0x0, 0x0, 0xc109bcb000?}, 0xc0380ebb20)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/session/session.go:1954 +0x2d2
github.com/pingcap/tidb/pkg/session.(*session).ExecRestrictedSQL(0x2?, {0x70d3428?, 0xc0470354d0?}, {0x0?, 0x0?, 0xc006b916c0?}, {0xc08250ca80?, 0x0?}, {0x0, 0x0, ...})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/session/session.go:1958 +0x88
github.com/pingcap/tidb/pkg/domain.(*Domain).refreshMDLCheckTableInfo(0xc001b76288)
	/home/genius/project/src/github.com/pingcap/tidb/pkg/domain/domain.go:1027 +0x42a
github.com/pingcap/tidb/pkg/domain.(*Domain).loadSchemaInLoop(0xc001b76288, {0x70d3460, 0xc002164f00})
	/home/genius/project/src/github.com/pingcap/tidb/pkg/domain/domain.go:1185 +0xde
github.com/pingcap/tidb/pkg/domain.(*Domain).Start.func2()
	/home/genius/project/src/github.com/pingcap/tidb/pkg/domain/domain.go:1534 +0x25
github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1()
	/home/genius/project/src/github.com/pingcap/tidb/pkg/util/wait_group_wrapper.go:103 +0x5c
created by github.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run in goroutine 1
	/home/genius/project/src/github.com/pingcap/tidb/pkg/util/wait_group_wrapper.go:98 +0xae
```

Reply:
Not passing the ctx correctly cause the `loadSchemaInLoop` fail to exit, the stack is above